processor/src/trace/range/mod.rs

```diff
@@ -0,0 +1,55 @@
+use super::{Felt, FieldElement};
+
+use crate::range::AuxColumnHint;
+
+pub use vm_core::range::{P0_COL_IDX, P1_COL_IDX, T_COL_IDX, V_COL_IDX};
+
+#[cfg(test)]
+mod tests;
+
+// HELPER FUNCTIONS
+// ================================================================================================
+
+/// Builds the execution trace of the range checker's `p0` auxiliary column used for multiset
+/// checks. The running product is built up in the 8-bit section of the table and reduced in the
+/// 16-bit section of the table so that the starting and ending value are both one.
+pub fn build_aux_col_p0<E: FieldElement<BaseField = Felt>>(
+    aux_column: &mut [E],
+    aux_column_hints: &[AuxColumnHint],
+    rand_elements: &[E],
+    t_col: &[Felt],
+    v_col: &[Felt],
+) {
+    aux_column[0] = E::ONE;
+
+    // Build the execution trace of the 8-bit running product.
+    for (row_idx, hint) in aux_column_hints
+        .iter()
+        .enumerate()
+        .take(aux_column.len() - 1)
+    {
+        let t: E = t_col[row_idx].into();
+        let v: E = v_col[row_idx].into();
+        let alpha = rand_elements[0];
```

Comment:
nit: `alpha` is the same for every row - let's move this assignment out of the loop? (the compiler probably takes care of it, but still...).

Reply:
Yep, that's an oversight. Thanks